src/ast/CMakeLists.txt

```diff
@@ -20,6 +20,7 @@ add_library(ast STATIC
   passes/semantic_analyser.cpp
   passes/codegen_llvm.cpp
   passes/return_path_analyser.cpp
+  passes/pid_filter_analyser.cpp
```

Comment:
Naming flame alert :warning: 

I don't think `_analyser` is a good suffix in this context as the pass doesn't just analyze but also mutates the AST.

LLVM distinguishes [Analysis and Transform Passes](https://llvm.org/docs/Passes.html) so I think we should do the same as people may be already familiar with them. So, I'd go with `passes/pid_filter_transformer.cpp`.

Reply:
Renamed it to PidFilterPass. WDYT?